workers/unity/Assets/Gdk/Core/Components/Option.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+
+namespace Improbable.Gdk.Core
+{
+    public struct Option<T> : IEquatable<Option<T>>
+    {
+        private bool hasValue;
```

Comment:
Shouldn't this be a BlittableBool ?

Reply:
is there a reason why we have a private and a public version of the variable? The private one doesn't look like it should be readonly or final, so we can just make it public in general

Reply:
@zeroZshadow Yes, I derped here. Should be a blittablebool.
@jessicafalk Having public fields would allow users to tamper with the integrity of an option. You could e.g. set the payload while hasValue is false rather than using the designated interface for setting getting, clearing.